### PR TITLE
Issue #7852: Window Vectorisation

### PR DESCRIPTION
**Pull request by hawkfish**
* Reuse tree aggregations between frames. 
* Reduce fan-out from 64 to recommended 16.
* Fix intitialisation without combine. (Another victory for vector size 2.)

Performance gains:

| Change | Time (s) |
| :--- | ---: |
| Baseline | 14.817 |
| Reuse | 10.031 |
| Fan Out | 8.350 |

### CODE REVIEW

**Comment by hawkfish** [comment]
Viktor and I also profiled the two changes independently.

| Fan Out | No Reuse | Reuse | Gain |
| :--- | ---: | ---: | ---: |
| 16 | 10.654 | 8.350 | 127% |
| 64 | 14.817 | 10.031 | 147% |
| Gain | 139% | 120% |

File: test/sql/window/test_parallel_window.test_slow [hunk 1]
```diff
@@ -2,24 +2,12 @@
 # description: Hashing and parallel execution
 # group: [window]
 
-#statement ok
-#PRAGMA enable_verification
```
Comment (Mytherin) [inline test/sql/window/test_parallel_window.test_slow:6]: Are these changes intentional?

File: src/execution/window_segment_tree.cpp [hunk 2]
```diff
@@ -500,6 +495,37 @@ void WindowSegmentTree::Evaluate(const idx_t *begins, const idx_t *ends, Vector
 	}
 	FlushStates(true);
 
+	//	Seggond pass: aggregate the ragged leaves
```
Comment (Mytherin) [inline src/execution/window_segment_tree.cpp:498]: Second? :)

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! LGTM - small comment:

File: test/sql/window/test_parallel_window.test_slow (hunk 1 above)
Comment (hawkfish) [reply test/sql/window/test_parallel_window.test_slow:6]: Yes I just deleted some commented out lines. There were failures here and I wanted to be sure I knew what was actually being used.

File: src/execution/window_segment_tree.cpp (hunk 2 above)
Comment (hawkfish) [reply src/execution/window_segment_tree.cpp:498]: It is now.  My push I forgot to fix the comments.

**Comment by Mytherin** [comment]
Thanks!
